extra/utils.py

```diff
@@ -209,12 +209,12 @@ def get_child(parent, key):
       obj = getattr(obj, k)
   return obj
 
-def _tree(lazydata):
-  if type(lazydata).__name__ == "LazyBuffer": return [f"━━ realized {lazydata.dtype.name} {lazydata.shape}"] if (lazydata.realized) else _tree(lazydata.op)
-  if len(lazydata.src) == 0: return [f"━━ {lazydata.op.name} {lazydata.arg if lazydata.arg else ''}"]
-  lines = [f"━┳ {lazydata.op.name} {lazydata.arg if lazydata.arg else ''}"]
+def _tree(lazydata, prefix=""):
+  if type(lazydata).__name__ == "LazyBuffer": return [f"━━ realized {lazydata.dtype.name} {lazydata.shape}"] if (lazydata.realized) else _tree(lazydata.op, "LB ")
+  if len(lazydata.src) == 0: return [f"━━ {prefix}{lazydata.op.name} {lazydata.arg if lazydata.arg else ''}"]
+  lines = [f"━┳ {prefix}{lazydata.op.name} {lazydata.arg if lazydata.arg else ''}"]
   childs = [_tree(c) for c in lazydata.src[:]]
   for c in childs[:-1]: lines += [f" ┣{c[0]}"] + [f" ┃{l}" for l in c[1:]]
   return lines + [" ┗"+childs[-1][0]] + ["  "+l for l in childs[-1][1:]]
 
-def print_tree(tensor:Tensor):print("\n".join([f"{str(i).rjust(3)} {s}" for i,s in enumerate(_tree(tensor.lazydata))]))
\ No newline at end of file
+def print_tree(tensor:Tensor):print("\n".join([f"{str(i).rjust(3)} {s}" for i,s in enumerate(_tree(tensor if type(tensor) != Tensor else tensor.lazydata))]))
```

Comment:
this should be `isinstance` not a comparison against `type()`

Reply:
```
a = Tensor.cat(*[Tensor([1]) for _ in range(4)])
print_tree(a)
```

Reply:
```
  0 ━┳ LB ADD 
  1  ┣━┳ ADD 
  2  ┃ ┣━┳ ADD 
  3  ┃ ┃ ┣━┳ LB PAD ((0, 3),)
  4  ┃ ┃ ┃ ┗━━ realized float (1,)
  5  ┃ ┃ ┗━┳ LB PAD ((1, 2),)
  6  ┃ ┃   ┗━━ realized float (1,)
  7  ┃ ┗━┳ LB PAD ((2, 1),)
  8  ┃   ┗━━ realized float (1,)
  9  ┗━┳ LB PAD ((3, 0),)
 10    ┗━━ realized float (1,)
```

Reply:
before it just doesnt have any 'LB'